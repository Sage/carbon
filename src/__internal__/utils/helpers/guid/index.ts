/**
 * Generates a random guid, useful for creating unique IDs.
 */
export default (): string => {
  function s4(): string {
    return Math.floor((1 + Math.random()) * 0x10000)
      .toString(16)
      .substring(1);
  }
  return `${s4()}${s4()}-${s4()}-${s4()}-${s4()}-${s4()}${s4()}${s4()}`;
};
