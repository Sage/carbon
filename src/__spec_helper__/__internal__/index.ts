import { configure } from "@testing-library/react";
import { enableFetchMocks } from "jest-fetch-mock";
import { setupMatchMediaMock } from "../mock-match-media";
import setupResizeObserverMock from "../mock-resize-observer";
import setupScrollToMock from "../mock-element-scrollto";
import "@testing-library/jest-dom";

enableFetchMocks();
setupResizeObserverMock();
setupMatchMediaMock();
setupScrollToMock();

configure({
  testIdAttribute:
    "data-role" /** Configure React Testing Library *ByTestId queries to use data-role tag */,
});
