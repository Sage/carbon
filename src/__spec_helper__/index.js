import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { setup } from './mock-match-media';

setup();
Enzyme.configure({ adapter: new Adapter() });
