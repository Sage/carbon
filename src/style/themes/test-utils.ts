const isObject = (obj: Record<string, unknown>) => {
  return typeof obj === "object" && obj !== null;
};

// eslint-disable-next-line import/prefer-default-export
export const assertIsSubset = (
  obj: Record<string, unknown>,
  comparison: Record<string, unknown>,
) => {
  if (!isObject(obj)) {
    // no further nesting
    return;
  }

  const objKeys = Object.keys(obj);
  const comparisonKeys = Object.keys(comparison);

  objKeys.forEach((key) => {
    // assert that keys are present
    expect(comparisonKeys).toContain(key);

    // repeat for nested objects
    assertIsSubset(
      obj[key] as Record<string, unknown>,
      comparison[key] as Record<string, unknown>,
    );
  });
};
