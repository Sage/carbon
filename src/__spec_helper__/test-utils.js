import { mount } from 'enzyme';

import { carbonThemeList } from '../style/themes';
import { mockMatchMedia } from './mock-match-media';

const isUpper = char => char.toUpperCase() === char;
const humpToDash = (acc, char) => `${acc}${isUpper(char) ? `-${char.toLowerCase()}` : char}`;

const toCSSCase = (str) => {
  return str.split('').reduce(humpToDash, '');
};

const assertStyleMatch = (styleSpec, component, opts) => {
  Object.entries(styleSpec).forEach(([attr, value]) => {
    expect(component).toHaveStyleRule(toCSSCase(attr), value, opts);
  });
};

const makeArrayKeys = n => [...Array(n).keys()];

const dispatchKeyPress = (code) => {
  const ev = new KeyboardEvent('keydown', { which: code });
  document.dispatchEvent(ev);
};

const keyMap = {
  UpArrow: '38',
  DownArrow: '40',
  RightArrow: '39',
  LeftArrow: '37',
  Enter: '13',
  Tab: '9',
  Space: '32',
  Escape: '27',
  End: '35',
  Home: '36',
  D: '68',
  E: '69',
  P: '80',
  Z: '90',
  1: '49'
};

const repeat = action => (n = 1) => makeArrayKeys(n).forEach(() => action());

const keyboard = Object.keys(keyMap).reduce((acc, key) => {
  acc[`press${key}`] = () => repeat(dispatchKeyPress(keyMap[key]));
  return acc;
}, {});

// Build an object of Enzyme simulate helpers
// e.g. simulate.keydown.pressTab(target, { shiftKey: true })
// e.g. simulate.keydown.pressEscape(target)
const keydown = Object.keys(keyMap).reduce((acc, key) => {
  acc[`press${key}`] = (target, { shiftKey } = { shiftKey: false }) => {
    target.simulate('keydown', { shiftKey, key, which: parseInt(keyMap[key], 0) });
  };
  return acc;
}, {});

const simulate = {
  keydown
};

const listFrom = wrapper => wrapper.find('ul');
const childrenFrom = node => node.children();

const hoverList = wrapper => (item) => {
  childrenFrom(listFrom(wrapper)).at(item).simulate('mouseover');
};

const simulateEvent = eventName => wrapper => wrapper.simulate(eventName);
const click = simulateEvent('click');

const selectedItemOf = wrapper => wrapper.state().selectedItem;
const isUnique = (val, index, self) => self.indexOf(val) === index;
const isSelectableGiven = nonSelectables => i => !nonSelectables.includes(i);

const selectedItemReducer = method => wrapper => (acc, i) => {
  method(wrapper)(i);
  return [...acc, selectedItemOf(wrapper)];
};

const arraysEqual = (arr1, arr2) => arr1.sort().join(',') === arr2.sort().join(',');

const assertCorrectTraversal = method => expect => ({ num, nonSelectables = [] }) => (wrapper) => {
  const array = makeArrayKeys(num);
  const validIndexes = array.filter(isSelectableGiven(nonSelectables));


  const selectedItem = selectedItemOf(wrapper);
  const indexesThatWereSelected = array
    .reduce(selectedItemReducer(method)(wrapper), [selectedItem])
    .filter(isUnique);
  expect(arraysEqual(validIndexes, indexesThatWereSelected)).toBeTruthy();
};

const assertKeyboardTraversal = assertCorrectTraversal(() => keyboard.pressDownArrow)(expect);
const assertHoverTraversal = assertCorrectTraversal(wrapper => hoverList(wrapper))(expect);

const carbonThemesJestTable = carbonThemeList.map(theme => [theme.name, theme]);

const spacingProps = [
  ['m', 'margin'],
  ['ml', 'marginLeft'],
  ['mr', 'marginRight'],
  ['mt', 'marginTop'],
  ['mb', 'marginBottom'],
  ['mx', 'marginLeft'],
  ['mx', 'marginRight'],
  ['my', 'marginTop'],
  ['my', 'marginBottom'],
  ['p', 'padding'],
  ['pl', 'paddingLeft'],
  ['pr', 'paddingRight'],
  ['pt', 'paddingTop'],
  ['pb', 'paddingBottom'],
  ['px', 'paddingLeft'],
  ['px', 'paddingRight'],
  ['py', 'paddingTop'],
  ['py', 'paddingBottom']
];

const getDefaultValue = (value) => {
  if (typeof value === 'number') {
    return `${value * 8}px`;
  }
  return value;
};

const testStyledSystemSpacing = (component, defaults, styleContainer) => {
  describe('default props', () => {
    const wrapper = mount(component());
    const StyleElement = styleContainer ? styleContainer(wrapper) : wrapper;

    it('should set the correct margins', () => {
      let margin;
      let marginLeft;
      let marginRight;
      let marginTop;
      let marginBottom;

      if (defaults) {
        margin = getDefaultValue(defaults.m || undefined);
        marginLeft = getDefaultValue(defaults.ml || defaults.mx || undefined);
        marginRight = getDefaultValue(defaults.mr || defaults.mx || undefined);
        marginTop = getDefaultValue(defaults.mt || defaults.my || undefined);
        marginBottom = getDefaultValue(defaults.mb || defaults.my || undefined);

        expect(assertStyleMatch(
          {
            margin,
            marginLeft,
            marginRight,
            marginTop,
            marginBottom
          },
          StyleElement
        ));
      } else {
        expect(StyleElement).not.toHaveStyleRule('marginLeft');
        expect(StyleElement).not.toHaveStyleRule('marginRight');
        expect(StyleElement).not.toHaveStyleRule('marginTop');
        expect(StyleElement).not.toHaveStyleRule('marginBottom');
        expect(StyleElement).not.toHaveStyleRule('margin');
      }
    });

    it('should set the correct paddings', () => {
      let padding;
      let paddingLeft;
      let paddingRight;
      let paddingTop;
      let paddingBottom;

      if (defaults) {
        padding = getDefaultValue(defaults.p || undefined);
        paddingLeft = getDefaultValue(defaults.pl || defaults.px || undefined);
        paddingRight = getDefaultValue(defaults.pr || defaults.px || undefined);
        paddingTop = getDefaultValue(defaults.pt || defaults.py || undefined);
        paddingBottom = getDefaultValue(defaults.pb || defaults.py || undefined);

        expect(assertStyleMatch(
          {
            padding,
            paddingLeft,
            paddingRight,
            paddingTop,
            paddingBottom
          },
          StyleElement
        ));
      } else {
        expect(StyleElement).not.toHaveStyleRule('paddingLeft');
        expect(StyleElement).not.toHaveStyleRule('paddingRight');
        expect(StyleElement).not.toHaveStyleRule('paddingTop');
        expect(StyleElement).not.toHaveStyleRule('paddingBottom');
        expect(StyleElement).not.toHaveStyleRule('padding');
      }
    });
  });

  describe.each(spacingProps)('when a custom spacing is specified using the "%s" styled system props',
    (styledSystemProp, propName) => {
      it(`then that ${propName} should have been set correctly`, () => {
        let wrapper = mount(component());

        const props = { [styledSystemProp]: 2 };
        wrapper = mount(component({ ...props }));

        expect(assertStyleMatch(
          { [propName]: '16px' },
          styleContainer ? styleContainer(wrapper) : wrapper
        ));
      });
    });
};

export {
  assertStyleMatch,
  toCSSCase,
  hoverList,
  selectedItemOf,
  childrenFrom,
  makeArrayKeys,
  keyboard,
  assertKeyboardTraversal,
  assertHoverTraversal,
  listFrom,
  click,
  simulate,
  carbonThemesJestTable,
  mockMatchMedia,
  testStyledSystemSpacing
};
