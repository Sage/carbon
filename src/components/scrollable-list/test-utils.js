import React from 'react';
import { makeArrayKeys } from '../../__spec_helper__/test-utils';
import ScrollableListItem from './scrollable-list-item.component';

const listItemReducer = ({ nonSelectables = [], customSelectables = [] }) => {
  // generate jsx for selectable list items, based on indexes in config
  return (acc, item, index) => {
    if (nonSelectables.includes(item)) return [...acc, <div />];
    if (customSelectables.includes(item)) return [...acc, <div isSelectable />];

    return [...acc, <ScrollableListItem id={ index } />];
  };
};

const renderListItems = (opts) => {
  return makeArrayKeys(opts.num).reduce(listItemReducer(opts), []);
};

export default renderListItems;
