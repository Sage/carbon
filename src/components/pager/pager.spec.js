import React from 'react';
import { mount } from 'enzyme';
import Immutable from 'immutable';
import 'jest-styled-components';
import { ThemeProvider } from 'styled-components';
import TestRenderer from 'react-test-renderer';
import I18n from 'i18n-js';
import guid from '../../utils/helpers/guid';
import { assertStyleMatch } from '../../__spec_helper__/test-utils';
import classicTheme from '../../style/themes/classic';
import mintTheme from '../../style/themes/mint';
import baseTheme from '../../style/themes/base';
import Pager from './pager.component';
import Dropdown from '../../__deprecated__/components/dropdown';

jest.mock('../../utils/helpers/guid');
guid.mockImplementation(() => 'guid-12345');

const pageSizeSelectionOptions = Immutable.fromJS([
  { id: '10', name: 10 },
  { id: '25', name: 25 },
  { id: '50', name: 50 }
]);

function render(props, renderType = mount) {
  return renderType(
    <ThemeProvider theme={ props.theme }>
      <Pager { ...props } />
    </ThemeProvider>
  );
}

describe('Pager', () => {
  const props = {
    currentPage: '1',
    totalRecords: 100,
    onPagination: () => true,
    pageSize: '10',
    showPageSizeSelection: true,
    pageSizeSelectionOptions
  };

  it('renders the Pager correctly with the Classic Theme', () => {
    const wrapper = render(
      { ...props, onPagination: () => true, theme: classicTheme },
      TestRenderer.create
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('renders the Pager correctly with the Mint Theme', () => {
    const wrapper = render(
      { ...props, onPagination: () => true, theme: mintTheme },
      TestRenderer.create
    );
    expect(wrapper).toMatchSnapshot();
  });

  describe('i18n', () => {
    const { translations } = I18n;
    const { locale } = I18n.locale;
    beforeAll(() => {
      I18n.translations = {
        ...translations,
        fr: {
          ...translations.fr,
          pager: {
            show: 'Spectacle',
            records: {
              one: 'article',
              zero: 'articles',
              other: 'articles'
            }
          }
        }
      };
    });

    afterAll(() => {
      I18n.translations = translations;
      I18n.locale = locale;
    });

    const getShow = wrapper => wrapper.find("div[data-element='page-select']").getDOMNode().parentElement.firstChild
      .textContent;
    const getRecords = wrapper => wrapper.find("div[data-element='page-select']").getDOMNode().parentElement.lastChild
      .textContent;
    const getTotalRecords = wrapper => wrapper.getDOMNode().lastChild.textContent;

    describe('default', () => {
      it('show', () => {
        const wrapper = render({ ...props, theme: mintTheme });
        expect(getShow(wrapper)).toBe('Show');
      });

      it('records', () => {
        expect(getRecords(render({ ...props, theme: mintTheme, pageSize: '100' }))).toBe('items');
        expect(getRecords(render({ ...props, theme: mintTheme, pageSize: '1' }))).toBe('item');
        expect(getRecords(render({ ...props, theme: mintTheme, pageSize: '0' }))).toBe('items');
        expect(getRecords(render({ ...props, theme: classicTheme, pageSize: '100' }))).toBe('records');
        expect(getRecords(render({ ...props, theme: classicTheme, pageSize: '1' }))).toBe('record');
        expect(getRecords(render({ ...props, theme: classicTheme, pageSize: '0' }))).toBe('records');
      });

      it('total records', () => {
        expect(getTotalRecords(render({ ...props, theme: mintTheme, totalRecords: 100 }))).toBe('100 items');
        expect(getTotalRecords(render({ ...props, theme: mintTheme, totalRecords: 1 }))).toBe('1 item');
        expect(getTotalRecords(render({ ...props, theme: mintTheme, totalRecords: 0 }))).toBe('0 items');
        expect(getTotalRecords(render({ ...props, theme: classicTheme, totalRecords: 100 }))).toBe('100 records');
        expect(getTotalRecords(render({ ...props, theme: classicTheme, totalRecords: 1 }))).toBe('1 record');
        expect(getTotalRecords(render({ ...props, theme: classicTheme, totalRecords: 0 }))).toBe('0 records');
      });
    });

    describe('fr', () => {
      beforeAll(() => {
        I18n.locale = 'fr';
      });

      it('show', () => {
        const wrapper = render({ ...props, theme: mintTheme });
        expect(getShow(wrapper)).toBe('Spectacle');
      });

      it('records', () => {
        expect(getRecords(render({ ...props, theme: mintTheme, pageSize: '100' }))).toBe('articles');
        expect(getRecords(render({ ...props, theme: mintTheme, pageSize: '1' }))).toBe('article');
        expect(getRecords(render({ ...props, theme: mintTheme, pageSize: '0' }))).toBe('articles');
      });

      it('total records', () => {
        expect(getTotalRecords(render({ ...props, theme: mintTheme, totalRecords: 100 }))).toBe('100 articles');
        expect(getTotalRecords(render({ ...props, theme: mintTheme, totalRecords: 1 }))).toBe('1 article');
        expect(getTotalRecords(render({ ...props, theme: mintTheme, totalRecords: 0 }))).toBe('0 articles');
      });
    });
  });

  describe('Classic theme', () => {
    it('has correct styles for PagerContainerStyles', () => {
      const wrapper = render({ ...props, theme: classicTheme }, TestRenderer.create).toJSON();
      assertStyleMatch({
        padding: '3px 16px',
        fontSize: '14px',
        backgroundColor: '#F2F4F5'
      }, wrapper);
    });
  });

  describe('DLS theme', () => {
    describe('Pager styling', () => {
      it('matches the expected style', () => {
        const wrapper = render(
          {
            ...props,
            theme: mintTheme
          },
          mount
        );

        assertStyleMatch({
          padding: '9px 24px',
          fontSize: '13px',
          border: `1px solid ${baseTheme.table.selected}`,
          backgroundColor: baseTheme.table.zebra,
          display: 'flex',
          justifyContent: 'space-between',
          alignItems: 'center',
          borderTopWidth: '0'
        }, wrapper);
      });
    });

    describe('Size Selector', () => {
      it('navigates correctly on page size update', () => {
        const onPagination = jest.fn();
        const wrapper = render(
          {
            ...props,
            currentPage: '4',
            onPagination,
            theme: mintTheme
          },
          mount
        );
        const dropdown = wrapper.find(Dropdown);
        dropdown.instance().selectValue('25', '25');
        expect(onPagination).toHaveBeenCalledWith('1', '25', 'size');
      });
    });
  });
});
