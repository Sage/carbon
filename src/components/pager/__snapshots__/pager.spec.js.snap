// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pager renders the Pager correctly with the Classic Theme 1`] = `
.c3 {
  display: inline-block;
  position: relative;
  color: rgba(0,0,0,0.65);
  background-color: transparent;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 24px;
  width: 24px;
  color: rgba(0,0,0,0.85);
}

.c3:hover {
  color: rgba(0,0,0,0.90);
  background-color: transparent;
}

.c3::before {
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-family: CarbonIcons;
  content: "\\e910";
  font-size: 16px;
  font-style: normal;
  font-weight: normal;
  line-height: 16px;
  vertical-align: middle;
  display: block;
}

.common-input__label--help .c3 {
  width: auto;
  height: auto;
}

.c3::before {
  font-size: 16px;
  font-style: normal;
  font-weight: normal;
  line-height: 16px;
  vertical-align: middle;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 9px 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-top-width: 0;
  font-size: 13px;
  border: 1px solid #D8E0E3;
  background-color: #F9FAFB;
  border-top: none;
  padding: 3px 16px;
  font-size: 14px;
  background-color: #F2F4F5;
}

.c0 .common-input__input:active,
.c0 .common-input__input:hover {
  border-color: #CCD6DA;
}

.c0 .common-input__input:active {
  border-color: #265BC7;
}

.c0 .common-input__input:hover,
.c0 .common-input__input:focus {
  border-color: #99adb6;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 30%;
  -ms-flex: 1 1 30%;
  flex: 1 1 30%;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c1 .carbon-dropdown {
  width: 55px;
  margin: 0 4px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4.c4 .c10 {
  padding: 0;
  margin: 0 4px;
  line-height: 24px;
  min-height: 24px;
}

.c4.c4 .c9 {
  text-align: center;
  height: 24px;
}

.c4 .carbon-number__input {
  width: 35px;
  height: 31px;
  padding: 0;
  margin: 0 4px;
  line-height: 24px;
  text-align: center;
}

.c5 {
  display: inline-block;
  -webkit-transform: rotate(90deg) scale(1.4);
  -ms-transform: rotate(90deg) scale(1.4);
  transform: rotate(90deg) scale(1.4);
  cursor: pointer;
  margin-right: 8px;
  opacity: 0.3;
  cursor: not-allowed;
}

.c7 {
  display: inline-block;
  -webkit-transform: rotate(90deg) scale(1.4);
  -ms-transform: rotate(90deg) scale(1.4);
  transform: rotate(90deg) scale(1.4);
  cursor: pointer;
  margin-right: 8px;
  -webkit-transform: rotate(-90deg) scale(1.4);
  -ms-transform: rotate(-90deg) scale(1.4);
  transform: rotate(-90deg) scale(1.4);
  margin-right: 0;
  margin-left: 8px;
}

.c6 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 30%;
  -ms-flex: 1 1 30%;
  flex: 1 1 30%;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

<div
  className="c0"
  data-component="pager"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <span>
        Show
      </span>
      <div
        className="carbon-dropdown common-input--with-icon common-input"
        data-component="dropdown"
        data-element="page-select"
      >
        <div
          className="common-input__field"
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <input
            autoComplete="off"
            className="carbon-dropdown__input common-input__input"
            data-element="input"
            id="guid-12345"
            name={null}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            readOnly={true}
            value={10}
          />
          <label
            htmlFor="guid-12345"
          >
            <span
              className="carbon-input-icon c3"
              data-component="icon"
              data-element="dropdown"
              disabled={false}
              fontSize="small"
              type="dropdown"
            />
          </label>
        </div>
        <input
          data-element="hidden-input"
          readOnly={true}
          type="hidden"
          value="10"
        />
      </div>
      <span>
        records
      </span>
    </div>
  </div>
  <div
    className="c4"
  >
    <div
      className="c5"
      disabled={true}
    >
      <span
        className="c3"
        data-component="icon"
        data-element="dropdown"
        disabled={false}
        fontSize="small"
        onClick={[Function]}
        type="dropdown"
      />
    </div>
    <span
      className="c6"
    >
      Page 
    </span>
    <label>
      <div
        className="carbon-number carbon-pager__current-page carbon-pager__current-page common-input"
        data-component="number"
        data-element="current-page"
      >
        <div
          className="common-input__field"
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <input
            autoComplete="off"
            className="carbon-number__input common-input__input"
            data-element="input"
            id="guid-12345"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            value="1"
          />
        </div>
      </div>
    </label>
    <span
      className="c6"
    >
       of 
      10
    </span>
    <div
      className="c7"
      disabled={false}
    >
      <span
        className="c3"
        data-component="icon"
        data-element="dropdown"
        disabled={false}
        fontSize="small"
        onClick={[Function]}
        type="dropdown"
      />
    </div>
  </div>
  <div
    className="c8"
  >
    100
     
    records
  </div>
</div>
`;

exports[`Pager renders the Pager correctly with the Mint Theme 1`] = `
.c14 {
  background: transparent;
  border: none;
  color: rgba(0,0,0,0.9);
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  font-size: 14px;
  outline: none;
  width: 30px;
}

.c14::-webkit-input-placeholder {
  color: rgba(0,0,0,0.3);
}

.c14::-moz-placeholder {
  color: rgba(0,0,0,0.3);
}

.c14:-ms-input-placeholder {
  color: rgba(0,0,0,0.3);
}

.c14::placeholder {
  color: rgba(0,0,0,0.3);
}

.c14:invalid,
.c14:required {
  box-shadow: none;
}

.c3 {
  display: inline-block;
  position: relative;
  color: rgba(0,0,0,0.65);
  background-color: transparent;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 24px;
  width: 24px;
}

.c3:hover {
  color: rgba(0,0,0,0.90);
  background-color: transparent;
}

.c3::before {
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-family: CarbonIcons;
  content: "\\e910";
  font-size: 16px;
  font-style: normal;
  font-weight: normal;
  line-height: 16px;
  vertical-align: middle;
  display: block;
}

.c12 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  background: #fff;
  border: 1px solid #668491;
  box-sizing: border-box;
  cursor: text;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex: 0 0 70%;
  -ms-flex: 0 0 70%;
  flex: 0 0 70%;
  margin: 0;
  min-height: 40px;
  padding-left: 11px;
  padding-right: 11px;
}

.c12 .c13 {
  height: calc(40px - 4px);
}

.c12 input::-ms-clear {
  display: none;
}

.c12 input::-webkit-contacts-auto-fill-button {
  display: none!important;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 9px 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-top-width: 0;
  font-size: 13px;
  border: 1px solid #D8E0E3;
  background-color: #F9FAFB;
  border-top: none;
}

.c0 .common-input__input:active,
.c0 .common-input__input:hover {
  border-color: #CCD6DA;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 30%;
  -ms-flex: 1 1 30%;
  flex: 1 1 30%;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c1 .carbon-dropdown {
  width: 55px;
  margin: 0 4px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c17.c17 .c11 {
  padding: 0;
  margin: 0 4px;
  line-height: 24px;
  min-height: 24px;
}

.c17.c17 .c13 {
  text-align: center;
  height: 24px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4.c4 .c11 {
  padding: 0;
  margin: 0 4px;
  line-height: 24px;
  min-height: 24px;
}

.c4.c4 .c13 {
  text-align: center;
  height: 24px;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 12px;
}

.c5 {
  padding: 0 4px;
  font-size: 13px;
  border-width: 0;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  background-color: transparent;
  cursor: pointer;
  color: rgba(0,0,0,0.74);
  color: rgba(0,0,0,0.55);
  cursor: not-allowed;
}

.c15 {
  padding: 0 4px;
  font-size: 13px;
  border-width: 0;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  background-color: transparent;
  cursor: pointer;
  color: rgba(0,0,0,0.74);
}

.c15:hover {
  color: rgba(0,0,0,0.90);
}

.c15:focus {
  outline: none;
  background-color: #FFDA7F;
  color: rgba(0,0,0,0.90);
}

.c7 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 30%;
  -ms-flex: 1 1 30%;
  flex: 1 1 30%;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c9 + .c8 {
  margin-top: 16px;
}

.c10 {
  display: block;
}

<div
  className="c0"
  data-component="pager"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <span>
        Show
      </span>
      <div
        className="carbon-dropdown common-input--with-icon common-input"
        data-component="dropdown"
        data-element="page-select"
      >
        <div
          className="common-input__field"
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <input
            autoComplete="off"
            className="carbon-dropdown__input common-input__input"
            data-element="input"
            id="guid-12345"
            name={null}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            readOnly={true}
            value={10}
          />
          <label
            htmlFor="guid-12345"
          >
            <span
              className="carbon-input-icon c3"
              data-component="icon"
              data-element="dropdown"
              disabled={false}
              fontSize="small"
              type="dropdown"
            />
          </label>
        </div>
        <input
          data-element="hidden-input"
          readOnly={true}
          type="hidden"
          value="10"
        />
      </div>
      <span>
        items
      </span>
    </div>
  </div>
  <div
    className="c4"
  >
    <button
      className="c5"
      disabled={true}
    >
      First
    </button>
    <button
      className="c5"
      disabled={true}
    >
      Previous
    </button>
    <div
      className="c6"
    >
      <span
        className="c7"
      >
        Page 
      </span>
      <label>
        <div
          className="c8 c9"
          data-element="current-page"
        >
          <div
            className="c10"
          >
            <div
              className="c11 c12"
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              role="presentation"
              size="medium"
            >
              <input
                aria-invalid={false}
                className="c13 c14"
                data-element="input"
                id="guid-12345"
                name="guid-12345"
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                size="medium"
                type="text"
                value="1"
              />
            </div>
          </div>
        </div>
      </label>
      <span
        className="c7"
      >
         of 
        10
      </span>
    </div>
    <button
      className="c15"
      disabled={false}
      onClick={[Function]}
    >
      Next
    </button>
    <button
      className="c15"
      disabled={false}
      onClick={[Function]}
    >
      Last
    </button>
  </div>
  <div
    className="c16"
  >
    100
     
    items
  </div>
</div>
`;
